manifests/config.pp

```diff
@@ -29,10 +29,16 @@
   $version = $::nexus::version,
 ) {
 
-  if $version !~ /\d.*/ or versioncmp($version, '3.0.0') >= 0 {
-    $conf_path = 'etc/org.sonatype.nexus.cfg'
-  } else {
-    $conf_path = 'conf/nexus.properties'
+  case $version {
```

Comment:
Let's implement the same logic, but continue using versioncmp instead of regexes.
```puppet
if $version !~ /\d.*/ or versioncmp($version, '3.1.0') >= 0 {
  $conf_path = 'etc/nexus.properties'
}
elsif versioncmp($version, '3.0.0') >= 0 {
  $conf_path = 'etc/org.sonatype.nexus.cfg'
}
else {
  $conf_path = 'conf/nexus.properties'
}
```